### DISCUSSION TITLE
REGEX Option Syntax Issue with Case

### DISCUSSION THREAD

**Discussion by RussCannon** [ksh93/ksh#647] [Q&A] [answered]
Hello.  My name is Russ, and I am new to this community since I have only recently discovered it.  I have been writing Korn Shell scripts since 1990 and am very pleased to find this project.

Since the issue came up in r/ksh about EREs in globbing, I have been experimenting with REGEX in various contexts and have run across a curious behavior in case blocks.

This bit of code:

```
typeset BOGOSITY='yAdA123YaDa'
case ${BOGOSITY} in
 ~(i)yada~(E)\d{3}yada) printf 'YES\n';;
 *) printf 'NO\n';;
esac
```

returns the following error in u+ and u+m:

```
syntax error at line 4: `}yada' unexpected
```

But, the following prints 'YES':

```
typeset BOGOSITY='yAdA123YaDa'
case ${BOGOSITY} in
 ''~(i)yada~(E)\d{3}yada) printf 'YES\n';;
 *) printf 'NO\n';;
esac
```

Why would the first not work, and why would an empty string prefixed to the pattern make it work?

Cheers,
Russ

**Comment by McDutchie** [comment, marked as answer]
Welcome. :)

I think you found another bug. It makes no sense that these should be different.

**Reply by RussCannon** [reply]
So, does this need to become an issue?

**Reply by McDutchie** [reply]
Yes, I've made an issue of it now. See #649

**Comment by JohnoKing** [comment]
An initial set of testing reveals the syntax error was introduced by the changes made in ksh93t 2008-10-31. Subsequently, this is what the changelog for that version says:
https://github.com/ksh93/ksh/blob/cfba37d35663ddce71f728c72d51a940e38ac808/src/cmd/ksh93/RELEASE#L1225-L1229
